### PR TITLE
Closes #1694: Add granular class selector to search block Submit button

### LINKED ISSUE #1694: Add more granular CSS selector for Search button
> Looks fine to me from both the anonymous and authenticated roles on desktop & mobile.
> 
> In order to prevent further issues with any class/ID changes that happen down the road, it might be worth while to apply a more unique class like `az-search-block` under the block configuration settings, and then to adjust the CSS specificity accordingly. How do we feel about making that kind of change to improve long-term maintenance?
> 

If we'd like to use a more specific class, we might go ahead and make that change in the [search block form twig](https://github.com/az-digital/az_quickstart/blob/main/themes/custom/az_barrio/templates/forms/form--search-block-form.html.twig) template. @akslay pointed out that if we make the change in that file instead of in the block configuration settings, people wouldn't be able to delete that class (accidentally or intentionally) which is required for our base styles.

_Originally posted by @bberndt-uaz in https://github.com/az-digital/az_quickstart/issues/1682#issuecomment-1162203502_

### PR DESCRIPTION
**Pull request by cpoff**
The submit box on the search bar in the main nav was using an obscure CSS selector, so I added `az-search-block` class and applied it to the main CSS sheet. This change should maintain existing functionality and give us a way to style that block specifically without breaking other placements.

## Description
Added class `az-search-block` to style.css and also to the Search block using the Block Configure tool (/admin/structure/block/manage/az_barrio_search?destination=/admin/structure/block)

## How to test
Go to arizona.edu homepage and search. When it takes you to the Search page, search again. The submit button should be white on pageload, and red when you hover.

Off canvas considerations.

## Types of changes
Added new CSS class to existing entry. Applied that class using the Drupal blocks UI.

### Arizona Quickstart (install profile, custom modules, custom theme)
- Patch release changes
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
- Minor release changes
   - [ ] New feature
   - [x] Breaking or visual change to existing behavior
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- [ ] Other or unknown

### Drupal core
- Patch release changes
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major or minor level update
- [ ] Other or unknown

### Drupal contrib projects
- Patch release changes
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major level update
- [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: themes/custom/az_barrio/css/style.css [hunk 1]
```diff
@@ -399,16 +399,21 @@ padding: 0.625rem .75rem;
...
   font-size: 20px;
   color: #8B0015;
   background-color: #ffffff;
   border: 1px solid #ced4da;
   padding: 0 8px;
 }
 
-form.search-block-form .input-group .input-group-append button:hover,
-form.search-block-form .input-group .input-group-append button:focus {
+.az-search-block button {
```
Comment (joshuasosa) [inline themes/custom/az_barrio/css/style.css:416]: These properties are a duplicate of lines above, 404-405. I'm not sure why the original lines were removed, but they should be re-added and this new line added to the selector list. Lines 402-406 here should be removed.

The `az-search-block` selector could be added to the button:hover and button:focus selectors to be consistent with `.az-search-block .search-block-form .input-group-append button`.

**Review by joshuasosa — COMMENTED** [review]
The new class, `az-search-block`, isn't being applied to the offcanvas search form. If the purpose of this issue/PR is to "improve long-term maintenance," the same class or a different new class should be applied in block.block.az_barrio_offcanvas_searchform.yml.

File: themes/custom/az_barrio/css/style.css (hunk 1 above)
Comment (trackleft) [inline themes/custom/az_barrio/css/style.css:416]: Like this?
```suggestion
.az-search-block form.search-block-form .input-group .input-group-append button:hover,
.az-search-block form.search-block-form .input-group .input-group-append button:focus {
```

File: themes/custom/az_barrio/css/style.css [hunk 2]
```diff
@@ -399,16 +399,21 @@ padding: 0.625rem .75rem;
...
 }
 
-.search-block-form .input-group-append button {
+form.search-block-form .input-group .input-group-append button:hover,
+form.search-block-form .input-group .input-group-append button:focus {
+    background-color: #8B0015;
+    color: #ffffff;
+}
+
+.az-search-block .search-block-form .input-group-append button {
```
Comment (trackleft) [inline themes/custom/az_barrio/css/style.css:408]:
```suggestion
.az-search-block form.search-block-form .input-group .input-group-append button {
```

Comment (joshuasosa) [reply themes/custom/az_barrio/css/style.css:408]: Yeah I think this and the other suggest change look good.

Comment (trackleft) [reply themes/custom/az_barrio/css/style.css:408]: Thanks!

**Review by tadean — APPROVED** [review]

**Review by jwmoore1 — APPROVED** [review]

**Pull request merged**